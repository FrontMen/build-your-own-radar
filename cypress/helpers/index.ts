export const dataCy = (value: string) => cy.get(`[data-cy=${value}]`);
export const dataTestId = (value: string) => cy.get(`[data-testid=${value}]`);

export const hexToRgb = (hex: string) => {
  const bigint = parseInt(hex.replace('#', ''), 16);
  const r = (bigint >> 16) & 255;
  const g = (bigint >> 8) & 255;
  const b = bigint & 255;

  return `rgb(${r}, ${g}, ${b})`;
}
