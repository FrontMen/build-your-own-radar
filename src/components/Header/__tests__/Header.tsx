import React from 'react';
import { Context as ResponsiveContext } from 'react-responsive';
import { Header } from '..';
import { withAllProviders } from 'test/helpers';

describe('Header', () => {
    /* test to check that the mobile and larger snapshot don't change unintentionally */
  test('matches the snapshot', () => {
    const { container: mobile } = withAllProviders(
      <ResponsiveContext.Provider value={{ width: 480 }}>
        <Header />
      </ResponsiveContext.Provider>,
    );

    expect(mobile).toMatchSnapshot();

    const { container: phablet } = withAllProviders(
      <ResponsiveContext.Provider value={{ width: 481 }}>
        <Header />
      </ResponsiveContext.Provider>,
    );

    expect(phablet).toMatchSnapshot();
  });
});
