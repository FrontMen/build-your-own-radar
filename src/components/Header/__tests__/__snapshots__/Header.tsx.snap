// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header matches the snapshot 1`] = `
.c0 {
  grid-area: header;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #fff;
  padding: 0px 8px;
  height: 138px;
}

.c1 {
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
}

.c2 {
  font-family: Montserrat,san-serif;
  font-weight: 600;
  line-height: 1.15;
  font-size: 1.5em;
  margin: 0;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c5 {
  border: 0;
  padding: 0;
  margin-left: 1em;
  background-color: transparent;
}

.c6 {
  width: 3em;
  height: 3em;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-basis: 100%;
  -ms-flex-preferred-size: 100%;
  flex-basis: 100%;
  -webkit-order: 3;
  -ms-flex-order: 3;
  order: 3;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  justify-items: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-family: Montserrat,san-serif;
  font-weight: 600;
  line-height: 1.15;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: 1em;
  color: inherit;
  width: 50%;
  height: 3em;
  padding: 0px 8px;
}

@media (min-width:481px) {
  .c0 {
    padding: 8px 32px;
  }
}

@media (min-width:1024px) {
  .c0 {
    height: 88px;
  }
}

@media (min-width:481px) {
  .c2 {
    font-size: 2.25em;
    font-weight: 700;
  }
}

@media (min-width:1024px) {
  .c3 {
    -webkit-order: unset;
    -ms-flex-order: unset;
    order: unset;
    -webkit-flex-basis: auto;
    -ms-flex-preferred-size: auto;
    flex-basis: auto;
  }
}

@media (min-width:481px) {
  .c4 {
    width: auto;
  }
}

<div>
  <div
    class="PageSlots__HeaderSlot-sc-1quj5xc-0 c0"
  >
    <a
      class="c1"
      href="/"
    >
      <h1
        class="c2"
      >
        Tech Radar
      </h1>
    </a>
    <div
      class="c3"
    >
      <a
        class="c4"
        href="/Frameworks, CMS & Programmeertalen"
      >
        Frameworks, CMS & Programmeertalen
      </a>
      <a
        class="c4"
        href="/Tooling en testing"
      >
        Tooling en testing
      </a>
      <a
        class="c4"
        href="/Platforms, infrastructure & Data"
      >
        Platforms, infrastructure & Data
      </a>
      <a
        class="c4"
        href="/Technieken"
      >
        Technieken
      </a>
    </div>
    <button
      class="c5"
    >
      <svg
        class="c6"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 512 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M443.5 420.2L336.7 312.4c20.9-26.2 33.5-59.4 33.5-95.5 0-84.5-68.5-153-153.1-153S64 132.5 64 217s68.5 153 153.1 153c36.6 0 70.1-12.8 96.5-34.2l106.1 107.1c3.2 3.4 7.6 5.1 11.9 5.1 4.1 0 8.2-1.5 11.3-4.5 6.6-6.3 6.8-16.7.6-23.3zm-226.4-83.1c-32.1 0-62.3-12.5-85-35.2-22.7-22.7-35.2-52.9-35.2-84.9 0-32.1 12.5-62.3 35.2-84.9 22.7-22.7 52.9-35.2 85-35.2s62.3 12.5 85 35.2c22.7 22.7 35.2 52.9 35.2 84.9 0 32.1-12.5 62.3-35.2 84.9-22.7 22.7-52.9 35.2-85 35.2z"
        />
      </svg>
    </button>
  </div>
</div>
`;

exports[`Header matches the snapshot 2`] = `
.c0 {
  grid-area: header;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #fff;
  padding: 0px 8px;
  height: 138px;
}

.c1 {
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
}

.c2 {
  font-family: Montserrat,san-serif;
  font-weight: 600;
  line-height: 1.15;
  font-size: 1.5em;
  margin: 0;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c5 {
  border: 0;
  padding: 0;
  margin-left: 1em;
  background-color: transparent;
}

.c6 {
  width: 3em;
  height: 3em;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-basis: 100%;
  -ms-flex-preferred-size: 100%;
  flex-basis: 100%;
  -webkit-order: 3;
  -ms-flex-order: 3;
  order: 3;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  justify-items: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-family: Montserrat,san-serif;
  font-weight: 600;
  line-height: 1.15;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: 1em;
  color: inherit;
  width: 50%;
  height: 3em;
  padding: 0px 8px;
}

@media (min-width:481px) {
  .c0 {
    padding: 8px 32px;
  }
}

@media (min-width:1024px) {
  .c0 {
    height: 88px;
  }
}

@media (min-width:481px) {
  .c2 {
    font-size: 2.25em;
    font-weight: 700;
  }
}

@media (min-width:1024px) {
  .c3 {
    -webkit-order: unset;
    -ms-flex-order: unset;
    order: unset;
    -webkit-flex-basis: auto;
    -ms-flex-preferred-size: auto;
    flex-basis: auto;
  }
}

@media (min-width:481px) {
  .c4 {
    width: auto;
  }
}

<div>
  <div
    class="PageSlots__HeaderSlot-sc-1quj5xc-0 c0"
  >
    <a
      class="c1"
      href="/"
    >
      <h1
        class="c2"
      >
        Tech Radar
      </h1>
    </a>
    <div
      class="c3"
    >
      <a
        class="c4"
        href="/Frameworks, CMS & Programmeertalen"
      >
        Frameworks, CMS & Programmeertalen
      </a>
      <a
        class="c4"
        href="/Tooling en testing"
      >
        Tooling en testing
      </a>
      <a
        class="c4"
        href="/Platforms, infrastructure & Data"
      >
        Platforms, infrastructure & Data
      </a>
      <a
        class="c4"
        href="/Technieken"
      >
        Technieken
      </a>
    </div>
    <button
      class="c5"
    >
      <svg
        class="c6"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 512 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M443.5 420.2L336.7 312.4c20.9-26.2 33.5-59.4 33.5-95.5 0-84.5-68.5-153-153.1-153S64 132.5 64 217s68.5 153 153.1 153c36.6 0 70.1-12.8 96.5-34.2l106.1 107.1c3.2 3.4 7.6 5.1 11.9 5.1 4.1 0 8.2-1.5 11.3-4.5 6.6-6.3 6.8-16.7.6-23.3zm-226.4-83.1c-32.1 0-62.3-12.5-85-35.2-22.7-22.7-35.2-52.9-35.2-84.9 0-32.1 12.5-62.3 35.2-84.9 22.7-22.7 52.9-35.2 85-35.2s62.3 12.5 85 35.2c22.7 22.7 35.2 52.9 35.2 84.9 0 32.1-12.5 62.3-35.2 84.9-22.7 22.7-52.9 35.2-85 35.2z"
        />
      </svg>
    </button>
  </div>
</div>
`;
